src/python/pants/backend/java/compile/javac_binary.py

```diff
@@ -25,14 +25,24 @@ class JavacBinary:
 
 @rule
 async def setup_javac_binary(coursier: Coursier, javac: JavacSubsystem) -> JavacBinary:
-    javac_wrapper_script = textwrap.dedent(
-        f"""\
-        set -eux
-        javac_path="$({coursier.coursier.exe} java-home --jvm {javac.options.jdk})/bin/javac"
-        /bin/mkdir -p {JavacBinary.classfiles_relpath}
-        exec "${{javac_path}}" "$@"
-        """
-    )
+    if javac.options.jdk == "system":
+        javac_wrapper_script = textwrap.dedent(
+            f"""\
+            set -eux
+            javac_path="$({coursier.coursier.exe} java-home --system-jvm)/bin/javac"
+            /bin/mkdir -p {JavacBinary.classfiles_relpath}
+            exec "${{javac_path}}" "$@"
+            """
+        )
```

Comment:
One thing that would harden this significantly would be to invoke a `Process(.., cache_scope=ProcessCacheScope.PER_RESTART_SUCCESSFUL)` here that dumped out the version that is actually selected by `Coursier`. You would then include that information as a comment or something in this script, which would cause the script's digest to change along with the actually-selected system JDK.

Reply:
Done

Reply:
Done.